Unstable Jenkins builds

The build log (latest: http://build.ros.org/job/Mdev__exotica__ubuntu_bionic_amd64/33/consoleFull) contains:

> ```
> 18:34:38 [WARNINGS] Plug-in Result: Unstable - <a href="warningsResult">20 warnings</a> exceed the threshold of 0 by 20
> 18:34:38 Build step 'Scan for compiler warnings' changed build result to UNSTABLE
> ```

These 20 compiler warnings are caused by deprecated throw specifications (`warning: dynamic exception specifications are deprecated in C++11 [-Wdeprecated]`) in the `pinocchio` library:
```C++
17:53:23 In file included from /tmp/ws/src/exotica/exotations/dynamics_solvers/exotica_pinocchio_dynamics_solver/include/exotica_pinocchio_dynamics_solver/pinocchio_dynamics_solver.h:39:0,
17:53:23                  from /tmp/ws/src/exotica/exotations/dynamics_solvers/exotica_pinocchio_dynamics_solver/src/pinocchio_dynamics_solver.cpp:30:
17:53:23 /opt/ros/melodic/include/pinocchio/parsers/urdf.hpp:33:44: warning: dynamic exception specifications are deprecated in C++11 [-Wdeprecated]
17:53:23                 const bool verbose = false) throw (std::invalid_argument);
17:53:23                                             ^~~~~
17:53:23 /opt/ros/melodic/include/pinocchio/parsers/urdf.hpp:48:44: warning: dynamic exception specifications are deprecated in C++11 [-Wdeprecated]
17:53:23                 const bool verbose = false) throw (std::invalid_argument);
17:53:23                                             ^~~~~
17:53:23 /opt/ros/melodic/include/pinocchio/parsers/urdf.hpp:103:5: warning: dynamic exception specifications are deprecated in C++11 [-Wdeprecated]
17:53:23      throw (std::invalid_argument);
17:53:23      ^~~~~
17:53:23 /opt/ros/melodic/include/pinocchio/parsers/urdf.hpp:120:5: warning: dynamic exception specifications are deprecated in C++11 [-Wdeprecated]
17:53:23      throw (std::invalid_argument);
17:53:23      ^~~~~
17:53:23 /opt/ros/melodic/include/pinocchio/parsers/urdf.hpp:149:5: warning: dynamic exception specifications are deprecated in C++11 [-Wdeprecated]
17:53:23      throw (std::invalid_argument);
17:53:23      ^~~~~
17:53:23 /opt/ros/melodic/include/pinocchio/parsers/urdf.hpp:176:5: warning: dynamic exception specifications are deprecated in C++11 [-Wdeprecated]
17:53:23      throw (std::invalid_argument)
17:53:23      ^~~~~
17:53:23 /opt/ros/melodic/include/pinocchio/parsers/urdf.hpp:208:5: warning: dynamic exception specifications are deprecated in C++11 [-Wdeprecated]
17:53:23      throw (std::invalid_argument);
17:53:23      ^~~~~
17:53:23 /opt/ros/melodic/include/pinocchio/parsers/urdf.hpp:235:5: warning: dynamic exception specifications are deprecated in C++11 [-Wdeprecated]
17:53:23      throw (std::invalid_argument)
17:53:23      ^~~~~
17:53:23 In file included from /opt/ros/melodic/include/pinocchio/parsers/urdf.hpp:245:0,
17:53:23                  from /tmp/ws/src/exotica/exotations/dynamics_solvers/exotica_pinocchio_dynamics_solver/include/exotica_pinocchio_dynamics_solver/pinocchio_dynamics_solver.h:39,
17:53:23                  from /tmp/ws/src/exotica/exotations/dynamics_solvers/exotica_pinocchio_dynamics_solver/src/pinocchio_dynamics_solver.cpp:30:
17:53:23 /opt/ros/melodic/include/pinocchio/parsers/urdf/model.hxx:194:36: warning: dynamic exception specifications are deprecated in C++11 [-Wdeprecated]
17:53:23                       bool verbose) throw (std::invalid_argument)
17:53:23                                     ^~~~~
17:53:23 /opt/ros/melodic/include/pinocchio/parsers/urdf/model.hxx:539:46: warning: dynamic exception specifications are deprecated in C++11 [-Wdeprecated]
17:53:23                           const bool verbose) throw (std::invalid_argument)
17:53:23                                               ^~~~~
17:53:23 /opt/ros/melodic/include/pinocchio/parsers/urdf/model.hxx:563:46: warning: dynamic exception specifications are deprecated in C++11 [-Wdeprecated]
17:53:23                           const bool verbose) throw (std::invalid_argument)
17:53:23                                               ^~~~~
17:53:23 /opt/ros/melodic/include/pinocchio/parsers/urdf/model.hxx:616:7: warning: dynamic exception specifications are deprecated in C++11 [-Wdeprecated]
17:53:23        throw (std::invalid_argument)
17:53:23        ^~~~~
17:53:23 /opt/ros/melodic/include/pinocchio/parsers/urdf/model.hxx:636:5: warning: dynamic exception specifications are deprecated in C++11 [-Wdeprecated]
17:53:23      throw (std::invalid_argument)
17:53:23      ^~~~~
17:53:23 /opt/ros/melodic/include/pinocchio/parsers/urdf/model.hxx:657:43: warning: dynamic exception specifications are deprecated in C++11 [-Wdeprecated]
17:53:23                        const bool verbose) throw (std::invalid_argument)
17:53:23                                            ^~~~~
17:53:23 /opt/ros/melodic/include/pinocchio/parsers/urdf/model.hxx:676:43: warning: dynamic exception specifications are deprecated in C++11 [-Wdeprecated]
17:53:23                        const bool verbose) throw (std::invalid_argument)
17:53:23                                            ^~~~~
17:53:23 In file included from /opt/ros/melodic/include/pinocchio/parsers/urdf/geometry.hxx:10:0,
17:53:23                  from /opt/ros/melodic/include/pinocchio/parsers/urdf.hpp:246,
17:53:23                  from /tmp/ws/src/exotica/exotations/dynamics_solvers/exotica_pinocchio_dynamics_solver/include/exotica_pinocchio_dynamics_solver/pinocchio_dynamics_solver.h:39,
17:53:23                  from /tmp/ws/src/exotica/exotations/dynamics_solvers/exotica_pinocchio_dynamics_solver/src/pinocchio_dynamics_solver.cpp:30:
17:53:23 /opt/ros/melodic/include/pinocchio/parsers/utils.hpp:65:91: warning: dynamic exception specifications are deprecated in C++11 [-Wdeprecated]
17:53:23                                             const std::vector<std::string> & package_dirs) throw (std::invalid_argument)
17:53:23                                                                                            ^~~~~
17:53:24 In file included from /opt/ros/melodic/include/pinocchio/parsers/urdf.hpp:246:0,
17:53:24                  from /tmp/ws/src/exotica/exotations/dynamics_solvers/exotica_pinocchio_dynamics_solver/include/exotica_pinocchio_dynamics_solver/pinocchio_dynamics_solver.h:39,
17:53:24                  from /tmp/ws/src/exotica/exotations/dynamics_solvers/exotica_pinocchio_dynamics_solver/src/pinocchio_dynamics_solver.cpp:30:
17:53:24 /opt/ros/melodic/include/pinocchio/parsers/urdf/geometry.hxx:301:93: warning: dynamic exception specifications are deprecated in C++11 [-Wdeprecated]
17:53:24                                               const std::vector<std::string> & package_dirs) throw (std::invalid_argument)
17:53:24                                                                                              ^~~~~
17:53:24 /opt/ros/melodic/include/pinocchio/parsers/urdf/geometry.hxx:382:54: warning: dynamic exception specifications are deprecated in C++11 [-Wdeprecated]
17:53:24                              const GeometryType type) throw (std::invalid_argument)
17:53:24                                                       ^~~~~
17:53:24 In file included from /opt/ros/melodic/include/pinocchio/parsers/urdf.hpp:246:0,
17:53:24                  from /tmp/ws/src/exotica/exotations/dynamics_solvers/exotica_pinocchio_dynamics_solver/include/exotica_pinocchio_dynamics_solver/pinocchio_dynamics_solver.h:39,
17:53:24                  from /tmp/ws/src/exotica/exotations/dynamics_solvers/exotica_pinocchio_dynamics_solver/src/pinocchio_dynamics_solver.cpp:30:
17:53:24 /opt/ros/melodic/include/pinocchio/parsers/urdf/geometry.hxx:413:7: warning: dynamic exception specifications are deprecated in C++11 [-Wdeprecated]
17:53:24        throw(std::invalid_argument)
17:53:24        ^~~~~
17:53:24 /opt/ros/melodic/include/pinocchio/parsers/urdf/geometry.hxx:430:7: warning: dynamic exception specifications are deprecated in C++11 [-Wdeprecated]
17:53:24        throw(std::invalid_argument)
17:53:24        ^~~~~
```

To resolve these warnings, these throw specifiers need to be removed upstream via a PR or they need to be removed via downstream patches during the bloom release.

--- a/exotations/dynamics_solvers/exotica_pinocchio_dynamics_solver/include/exotica_pinocchio_dynamics_solver/pinocchio_dynamics_solver.h
+++ b/exotations/dynamics_solvers/exotica_pinocchio_dynamics_solver/include/exotica_pinocchio_dynamics_solver/pinocchio_dynamics_solver.h
@@ -32,11 +32,18 @@
 
 #include <exotica_core/dynamics_solver.h>
 
+/// TODO: remove this pragma once Pinocchio removes neutralConfiguration/
+/// and fixes their deprecation warnings. (Relates to #547)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+
 #include <pinocchio/algorithm/aba-derivatives.hpp>
 #include <pinocchio/algorithm/aba.hpp>
 #include <pinocchio/multibody/data.hpp>
 #include <pinocchio/multibody/model.hpp>
 #include <pinocchio/parsers/urdf.hpp>
+
+#pragma GCC diagnostic pop
 
 namespace exotica
 {

